src/ucp/wireup/address.c

```diff
@@ -684,37 +996,73 @@ ucp_address_unpack_iface_length(ucp_worker_h worker, const void *flags_ptr,
 
     if (!is_ep_addr) {
         *is_last_iface = *(uint8_t*)ptr & UCP_ADDRESS_FLAG_LAST;
+        extendable     = addr_version != UCP_ADDRESS_VERSION_V1;
+        max_value      = UCP_ADDRESS_IFACE_LEN_MASK;
+    } else {
+        extendable     = 0;
+        max_value      = UINT8_MAX;
     }
 
-    *addr_length = *(uint8_t*)ptr & UCP_ADDRESS_IFACE_LEN_MASK;
-
-    return UCS_PTR_TYPE_OFFSET(ptr, uint8_t);
+    return ucp_address_unpack_byte_extended(ptr, max_value, extendable,
+                                            addr_length);
 }
 
-uint64_t ucp_address_get_uuid(const void *address)
+static void ucp_address_pack_header_flags(uint8_t *address_header,
+                                          uint8_t addr_version, uint8_t flags)
 {
-    const uint8_t address_header = *(const uint8_t *)address;
-    if (address_header & UCP_ADDRESS_HEADER_FLAG_WORKER_UUID) {
-        return *(uint64_t*)UCS_PTR_TYPE_OFFSET(address, uint8_t);
+    if (addr_version == UCP_ADDRESS_VERSION_V1) {
+        *address_header |= (flags << UCP_ADDRESS_HEADER_FLAGS_SHIFT_V1);
     } else {
-        return UCP_ADDRESS_DEFAULT_WORKER_UUID;
+        address_header += 1;
+        *address_header = flags;
```

Comment:
maybe define a struct of { header, flags } for address_header_v2?

Reply:
ok

Reply:
imo, it adds code, but does not make really clearer (tried, but reverted)